src/main/java/edu/kis/powp/jobs2d/drivers/adapter/ControllerAdapter.java

```diff
@@ -1,18 +1,21 @@
 package edu.kis.powp.jobs2d.drivers.adapter;
 
-import edu.kis.legacy.drawer.panel.DrawPanelController;
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
-import edu.kis.powp.jobs2d.Job2dDriver;
+
+import javax.swing.*;
 
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
-	private int startX = 0, startY = 0;
-
-	public MyAdapter() {
+public class ControllerAdapter extends AbstractAdapter {
+	public ControllerAdapter(JPanel panel) {
 		super();
+		this.panel = panel;
+		super.initialize(panel);
+	}
+	public void activate() {
+		super.initialize(panel);
```

Comment:
solution inconsistent with the instructions: Dostęp do instancji DrawPanelController zintegrowanej z oknem aplikacji jest zdefiniowany w metodzie getDrawerController() w klasie DrawerFeature.